### DISCUSSION TITLE
Query parameters handling by SWR routes

### DISCUSSION THREAD

**Discussion by lewebsimple** [nuxt/nuxt#19719] [Questions]
I have a `/search/?s=blabla` route deployed on Vercel with the following `routeRules` config:
```
routeRules: {
  '/search/**': { swr: false },
  '/**': { swr: 60 },
},
```
It seems the search results page is always pre-rendered with an empty `s` query, then executes the query on the client side after it mounts, which produces a weird experience.

Am I missing something?

**Comment by leopoldkristjansson** [comment]
Experiencing the same for a page that has pagination.

```js
  routeRules: {
    '/blog': { swr: true },
  },
```

Locally, I can use `/blog?page=4` without issues, but in production (Vercel) I get hydration errors and a flash, first showing content from `/blog`, before rendering the correct content for `/blog?page=4`.

Turning off SWR for this route seems to fix this.

Would be interested in knowing if there are ways around this?
